ui/src/external/monaco.flux.server.ts

```diff
@@ -60,6 +69,46 @@ export class LSPServer {
     return this.send(initialize(this.currentMessageID))
   }
 
+  async rename(uri, position, newName): Promise<WorkspaceEdit> {
+    const response = (await this.send(
+      rename(this.currentMessageID, uri, position, newName)
+    )) as {result: WorkspaceEdit}
+
+    return response.result
+  }
+
+  async definition(uri, position): Promise<Location> {
+    const response = (await this.send(
```

Comment:
feels like a lot of copy pasta going on. i forget why we arent just putting the message definitions here?

Reply:
I do like having functions to build the messages remove a layer to this. i.e. the server maintains state like the id and the message builders can be pure functions :)